src/types/basic.rs

```diff
@@ -104,8 +161,8 @@ impl<T> Type<T> {
     fn upcast<T2: From<T>>(self) -> Type<T2> {
         match self {
             Type::Prim(t) => Type::Prim(t.into()),
-            Type::Extension(t) => Type::Extension(t),
-            Type::Alias(_) => todo!(),
+            Type::Extension(Tagged(t, _)) => Type::Extension(Tagged(t, PhantomData)),
```

Comment:
fwiw If we had the old Upcast, we could call that here, as proof we know this is OK, rather than bypassing the constructor check (so this code here moves inside the "trust" boundary whereas before it was outside)

Reply:
i think that's ok  - but if it wasn't we could call Tagged::new here and implement try_into

Reply:
Yah I'm not seriously suggesting we shouldn't trust this code. One should be careful about this boundary creeping but I think it should be stable enough soon.